File: src/main/java/net/benwoodworth/fastcraft/core/api/event/FcEvent.java
package net.benwoodworth.fastcraft.core.api.event;

import org.jetbrains.annotations.NotNull;

/**
 * Adapts an event from a native implementation.
 */
public interface FcEvent<TBase> {

    /**
     * The base of this event.
     */
    @NotNull
    TBase getBaseEvent();

    /**
     * Whether this event has been cancelled.
     */
    boolean isCancelled();

    /**
     * Raise the base event.
     */
    void raise();
}

File: src/main/kotlin/net/benwoodworth/fastcraft/core/api/event/FcEvent.kt
package net.benwoodworth.fastcraft.core.api.event

/**
 * Adapts an event from a native implementation.
 *
 * @param TBase The base event from the server platform.
 */
interface FcEvent<out TBase> {
    /**
     * The base of this event.
     */
    val baseEvent: TBase

    /**
     * Whether this event has been cancelled.
     */
    var cancelled: Boolean

    /**
     * Raise the base event.
     */
    fun raise()
}
